sdk/eventhub/Azure.Messaging.EventHubs/perf/Infrastructure/BufferedPublishPerfTest.cs

```diff
@@ -0,0 +1,247 @@
+        ///
+        protected string[] Partitions { get; private set; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="BufferedPublishPerfTest"/> class.
+        /// </summary>
+        ///
+        /// <param name="options">The set of options to consider for configuring the scenario.</param>
+        ///
+        public BufferedPublishPerfTest(TOptions options) : base(options)
+        {
+        }
+
+        /// <summary>
+        ///   Performs the tasks needed to initialize and set up the environment for the test scenario.
+        ///   This setup will take place once for each instance, running after the global setup has
+        ///   completed.
+        /// </summary>
+        ///
+        public override async Task SetupAsync()
+        {
+            await base.SetupAsync();
+
+            Scope = await EventHubScope.CreateAsync(Options.PartitionCount).ConfigureAwait(false);
+
+            var producerOptions = new EventHubBufferedProducerClientOptions
+            {
+                MaximumWaitTime = (Options.MaximumWaitTimeMilliseconds.HasValue) ? TimeSpan.FromMilliseconds(Options.MaximumWaitTimeMilliseconds.Value) : null,
+                MaximumEventBufferLengthPerPartition = Options.MaximumBufferLength,
+                MaximumConcurrentSendsPerPartition = Options.MaximumConcurrentSendsPerPartition
+            };
+
+            if (Options.MaximumConcurrentSends.HasValue)
+            {
+                producerOptions.MaximumConcurrentSends = Options.MaximumConcurrentSends.Value;
+            }
+
+            _producer = new EventHubBufferedProducerClient(TestEnvironment.EventHubsConnectionString, Scope.EventHubName, producerOptions);
+
+            // Create the handlers that call into the performance test infrastructure to
+            // report when errors and events are observed.
+
+            _producer.SendEventBatchFailedAsync += args =>
+            {
+                // Do not flag cancellation as a failure; it is expected for in-flight batches when
+                // the test run is cleaning up.
+
+                if (!typeof(OperationCanceledException).IsAssignableFrom(args.Exception.GetType()))
+                {
+                    ErrorRaised(args.Exception);
+                }
+
+                return Task.CompletedTask;
+            };
+
+            _producer.SendEventBatchSucceededAsync += args =>
+            {
+                for (var index = 0; index < args.EventBatch.Count; ++index)
+                {
+                    EventRaised();
+                }
+
+                return Task.CompletedTask;
+            };
+
+            // Read the available partitions and buffer a single events to establish the
+            // connection and link and start reading the buffers.
+
+            Partitions = await _producer.GetPartitionIdsAsync().ConfigureAwait(false);
+            await _producer.EnqueueEventsAsync(EventGenerator.CreateEvents(1)).ConfigureAwait(false);
```

Comment:
Buffering a single event to establish the connection seems unnecessary, since the existing `warmup` period should handle this.

Reply:
Fair; we used this pattern in the other tests but I suppose that's mostly because of a lack of understanding at the time.   I'll make a pass to tweak that across the tests.